src/app/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, inject } from '@angular/core';
+import { Breakpoints, BreakpointObserver } from '@angular/cdk/layout';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.css']
+})
+export class DashboardComponent {
+  private breakpointObserver = inject(BreakpointObserver);
+
+  /** Based on the screen size, switch from standard to one column per row */
+  cards = this.breakpointObserver.observe(Breakpoints.Handset).pipe(
+    map(({ matches }) => {
+      if (matches) {
+        return [
+          { title: 'Card 1', cols: 1, rows: 1 },
+          { title: 'Card 2', cols: 1, rows: 1 },
+          { title: 'Card 3', cols: 1, rows: 1 },
+          { title: 'Card 4', cols: 1, rows: 1 }
+        ];
+      }
+
+      return [
+        { title: 'Card 1', cols: 2, rows: 1 },
+        { title: 'Card 2', cols: 1, rows: 1 },
+        { title: 'Card 3', cols: 1, rows: 2 },
+        { title: 'Card 4', cols: 1, rows: 1 }
+      ];
+    })
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This logic is tightly coupled with the component and hard to test. Consider moving this logic to a service that can be tested independently. This will improve the modularity and maintainability of your code.

For example, you could create a `DashboardService` with a method `getCards()` that returns an Observable of cards. Then, in your component, you would simply subscribe to this Observable.

```typescript
export class DashboardService {
  constructor(private breakpointObserver: BreakpointObserver) {}

  getCards(): Observable<Card[]> {
    return this.breakpointObserver.observe(Breakpoints.Handset).pipe(
      map(({ matches }) => {
        if (matches) {
          return [
            { title: 'Card 1', cols: 1, rows: 1 },
            { title: 'Card 2', cols: 1, rows: 1 },
            { title: 'Card 3', cols: 1, rows: 1 },
            { title: 'Card 4', cols: 1, rows: 1 }
          ];
        }

        return [
          { title: 'Card 1', cols: 2, rows: 1 },
          { title: 'Card 2', cols: 1, rows: 1 },
          { title: 'Card 3', cols: 1, rows: 2 },
          { title: 'Card 4', cols: 1, rows: 1 }
        ];
      })
    );
  }
}
```

Then in your component:

```typescript
export class DashboardComponent {
  cards: Observable<Card[]>;

  constructor(private dashboardService: DashboardService) {
    this.cards = this.dashboardService.getCards();
  }
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->